### PR TITLE
Fix tensor printing in `trainer.test()`

### LINKED ISSUE #5107: Pretty print tensors in `trainer.test()`
## 🐛 Bug

When printing the results of `trainer.test()` (saying "DATALOADER:0 TEST RESULTS"), it shows "tensor(...)", sometimes also including the device, when just a number should be shown.

## Please reproduce using [the BoringModel and post here](https://colab.research.google.com/drive/1HvWVVTK8j2Nj52qU4Q4YCyzOm0_aLQF3?usp=sharing)

I copied the template and didn't modify it, as it already shows the issue: https://colab.research.google.com/drive/1_lBgEkebPAdoZ-045qGpjUG_E4IUL60w?usp=sharing

It shows this:

```
DATALOADER:0 TEST RESULTS
{'fake_test_acc': tensor(4.7004e-14)}
```

### Expected behavior

```
DATALOADER:0 TEST RESULTS
{'fake_test_acc': 4.7004e-14}
```

### PR DESCRIPTION
**Pull request by bryant1410**
## What does this PR do?

Fix printing tensors when showing test results, so it goes from this

```
DATALOADER:0 TEST RESULTS
{'fake_test_acc': tensor(4.7004e-14, device="cuda:0")}
```

to this

```
DATALOADER:0 TEST RESULTS
{'fake_test_acc': 4.7004e-14}
```

Fixes #5107 

## Before submitting
- [x] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together? Otherwise, we ask you to create a separate PR for every change.
- [x] Did you make sure to update the documentation with your changes?
- [ ] Did you write any new necessary tests? 
- [x] Did you verify new and existing tests pass locally with your changes?
- [x] If you made a notable change (that affects users), did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)?

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:  

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [ ] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified; _Bugfixes should be including in bug-fix release milestones (m.f.X) and features should be included in (m.X.b) releases._
 

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

File: pytorch_lightning/trainer/connectors/logger_connector/logger_connector.py [hunk 1]
```diff
@@ -256,7 +256,7 @@ def get_evaluate_epoch_results(self, test_mode):
             print('-' * 80)
             for result_idx, results in enumerate(self.eval_loop_results):
                 print(f'DATALOADER:{result_idx} TEST RESULTS')
-                pprint(results)
+                pprint({k: v.tolist() if isinstance(v, torch.Tensor) else v for k, v in results.items()})
```
Comment (SkafteNicki) [inline pytorch_lightning/trainer/connectors/logger_connector/logger_connector.py:259]: Calling `.tolist()` will result in scalar tensors being wrapped in a list:
{'test_loss': [1.1703]}
Instead we should probably call `item` for scalar tensors and `tolist` for non-scalar tensors:
```suggestion
                convert = lambda x: x.item() if x.numel()==1 else x.tolist()
                pprint({k: convert(v) if isinstance(v, torch.Tensor) else v for k, v in results.items()})
```

Comment (bryant1410) [reply pytorch_lightning/trainer/connectors/logger_connector/logger_connector.py:259]: I had actually checked for that and at least in PyTorch 1.7.1 `torch.tensor(1.1703).tolist()` doesn't return square brackets to me: `{'test_loss': 1.1703}`. What PyTorch version are you using?

Comment (SkafteNicki) [reply pytorch_lightning/trainer/connectors/logger_connector/logger_connector.py:259]: I am running torch v1.6. Can we do something to be backward compatible to earlier versions of pytorch (since lightning in general is backward compatible to pytorch v1.3)?

Comment (bryant1410) [reply pytorch_lightning/trainer/connectors/logger_connector/logger_connector.py:259]: @SkafteNicki I've tried with PyTorch 1.6 and 1.3 as well, and none of them return the square brackets. Can you please confirm `torch.tensor(1.1703).tolist()` gives you square brackets?

Comment (SkafteNicki) [reply pytorch_lightning/trainer/connectors/logger_connector/logger_connector.py:259]: I commented because this simple example will fail:
```torch.randn(1).tolist() 
>>> [0.2993074953556061]
``` 
Maybe it is to artificial, but it seems that `tolist` only works when the shape of the tensor is `torch.Size([])` and not in general when it is a 1 element tensor.

**Review by tchaton — APPROVED** [review]

File: pytorch_lightning/trainer/connectors/logger_connector/logger_connector.py [hunk 2]
```diff
@@ -256,7 +256,8 @@ def get_evaluate_epoch_results(self, test_mode):
             print('-' * 80)
             for result_idx, results in enumerate(self.eval_loop_results):
                 print(f'DATALOADER:{result_idx} TEST RESULTS')
-                pprint(results)
+                pprint({k: (v.item() if v.numel() == 1 else v.tolist()) if isinstance(v, torch.Tensor) else v
+                        for k, v in results.items()})
```
Comment (justusschock) [inline pytorch_lightning/trainer/connectors/logger_connector/logger_connector.py:260]: Just thinking if it is worth to actually do our own pprint, that does this kind of unwrapping for tensors so that we can simply reuse this one (also nice for user)

**Review by justusschock — APPROVED** [review]

**Review by SkafteNicki — APPROVED** [review]

**Review by Borda — APPROVED** [review]

**Comment by Borda** [comment]
well, shall this be rather a feature?
